examples/dump_media/main.py

```diff
@@ -0,0 +1,201 @@
+            (
+                ["--dry_run"],
+                {
+                    "action": "store_true",
+                    "help": "Don't download, just create the directory structure locally",
+                },
+            ),
+        ]
+    )
+    print(args)
+    mediaDownloader = MediaDownloader(
+        args.config, args.root_dir, args.log_to_file, args.dry_run
+    )
+
+    print("Getting all media ids")
+    media_ids = mediaDownloader.get_all_media_ids_in_account()
+    print(f"Found {len(media_ids)} media in the account")
+
+    print("Getting collection and caption information for all media")
+    media_objects = mediaDownloader.extend_all_media_with_collection_and_captions(
+        media_ids
+    )
+    print("Extended all media with collection and caption information")
+
+    print("Start download")
+    mediaDownloader.download_all_media(media_objects)
+    print("Download is complete")
+
+
+def log_exception(func):
+    def inner(*args):
+        try:
+            return func(*args)
+        except Exception as e:
+            args[0].logger.error(e)
+            raise
+
+    return inner
+
+
+class MediaDownloader:
+    def __init__(self, config, root_dir, log_to_file, dry_run):
+        self.public_api_client = PublicAPIClient(config)
+        self.log_to_file = log_to_file
+        self.dry_run = dry_run
+        self.root_dir = Path(root_dir)
+        self.root_dir.mkdir(parents=True, exist_ok=True)
+
+        if log_to_file:
+            logfile_path = f"{root_dir}/media_dump_{datetime.now().strftime('%Y-%m-%d_%H:%M:%S')}.log"
+            log_format = "%(asctime)s - %(levelname)s - %(message)s"
+            logging.basicConfig(
+                filename=logfile_path, level=logging.INFO, format=log_format
+            )
+        self.logger = logging.getLogger(__name__)
+
+    @log_exception
+    def get_all_media_ids_in_account(self):
+        response = self.search_media(1)
+        media_ids = [
+            media["id"]
+            for media in response.json()["media"]
+            if MediaDownloader.is_media_downloadable(media)
+        ]
+        last_page = response.json()["meta"]["last_page"]
+        total_count = response.json()["meta"]["total_count"]
+        self.logger.info(f"There are {total_count} videos on {last_page} page.")
+        if last_page > 1:
+            for page in range(2, last_page + 1):
+                response = self.search_media(page)
+                media_ids.extend(
+                    [
+                        media["id"]
+                        for media in response.json()["media"]
+                        if MediaDownloader.is_media_downloadable(media)
+                    ]
+                )
+        logger.info(f"Found media ids: {media_ids}")
+        return media_ids
+
+    def search_media(self, page):
+        self.logger.info(f"Getting {page}. page")
+        response = self.public_api_client.request(
+            "get", "media/search", params={"page": page}
```

Comment:
how about sending per_page param as well and we can max out to 50. this way we reduce the number of the calls by getting bigger chunk of data:
```suggestion
            "get", "media/search", params={"page": page, "per_page": 50}
```

Reply:
good idea, will do!

Reply:
I don't think so. If you look at the server code, it already has some default:
```
      def search
...        query = TweedMediaManagement::Media.active
                                           .page(params[:page])
                                           .per(params[:per_page])
```

Reply:
Nevermind, makes sense.